specs/core/1_shard-data-chains.md

```diff
@@ -300,76 +302,52 @@ def is_valid_shard_block(beacon_blocks: Sequence[BeaconBlock],
             return True
 
     # Check slot number
-    assert candidate.slot >= PHASE_1_FORK_SLOT
-
-    # Check shard number
-    assert candidate.shard <= SHARD_COUNT
+    assert candidate.core.slot >= PHASE_1_FORK_SLOT
 
     # Check beacon block
-    beacon_block = beacon_blocks[candidate.slot]
-    assert candidate.beacon_block_root == signing_root(beacon_block)
-    assert beacon_block.slot <= candidate.slot
+    beacon_block = beacon_blocks[candidate.core.slot - candidate.core.slot % (SHARD_SLOTS_PER_BEACON_SLOT * SLOTS_PER_EPOCH)]
+    assert candidate.core.beacon_block_root == signing_root(beacon_block)
+    assert beacon_block.slot <= candidate.core.slot
 
     # Check state root
-    assert candidate.state_root == Hash()  # [to be removed in phase 2]
+    assert candidate.core.state_root == Hash()  # [to be removed in phase 2]
 
     # Check parent block
-    if candidate.slot == PHASE_1_FORK_SLOT:
-        assert candidate.parent_root == Hash()
+    if candidate.core.slot == PHASE_1_FORK_SLOT:
+        assert candidate.core.parent_root == Hash()
     else:
         parent_block = next(
-            (block for block in valid_shard_blocks if signing_root(block) == candidate.parent_root),
+            (block for block in valid_shard_blocks if hash_tree_root(block.core) == candidate.core.parent_root),
             None
         )
         assert parent_block is not None
-        assert parent_block.shard == candidate.shard
-        assert parent_block.slot < candidate.slot
-        assert signing_root(beacon_blocks[parent_block.slot]) == parent_block.beacon_chain_root
+        assert parent_block.core.slot < candidate.core.slot
+        assert signing_root(beacon_blocks[parent_block.core.slot]) == parent_block.core.beacon_chain_root
```

Comment:
```suggestion
        assert signing_root(beacon_blocks[compute_slot_from_shard_slot(parent_block.core.slot)]) == parent_block.core.beacon_chain_root
```

Reply:
This conditional statement should also be for the initial block in the shard chain but is skipped currently